test/controllers/thinkers_controller_test.rb

```diff
@@ -1,14 +1,49 @@
 require 'test_helper'
 
 class ThinkersControllerTest < ActionController::TestCase
+  def setup
+    @thinker = Thinker.create( name: "Example Thinker", email: "thinker@example.com")
+  end
+
   test "should get new" do
     get :new
     assert_response :success
   end
 
-  #test "should get destroy" do
-   # get :destroy
-   # assert_response :success
+  test "should get index" do
+    get :index
+    assert_response :success
+    assert_not_nil assigns(:thinkers)
+  end
+
+  test "should create thinker" do
+	  assert_difference('Thinker.count') do
+	  post :create, thinker: {name: 'Some name', email: 'email@example.com'}
+	  end
+ 
+  	assert_redirected_to thinker_path(assigns(:thinker))
+	end
+
+  test "should get show" do
+  	get :show, id: @thinker.id
+  	assert_response :success
+  end
+
+  test "should get edit" do
+  	get :edit, id: @thinker.id
+  	assert_response :success
+  end
+
+  #test "should update thinker" do
```

Comment:
I'm going to provide a complete example here in order to illustrate a few more suggestions. Here's my implementation:

```
  test "should update thinker" do
    params = {name: 'Another Thinker'}

    put :update, id: @thinker.id, thinker: params

    @thinker.reload
    assert_equal @thinker.name, params[:name]
    assert_redirected_to(assigns(:thinker))
  end
```

A few notes on what I've done:

1. The original failure message was due to the missing `params` field. (The error was `ActionController::ParameterMissing:         ActionController::ParameterMissing: param is missing or the value is empty: thinker`.)

2. Adding that caused a different failure: `Expected response to be a <success>, but was <302>`. This was because the 'success' assertion expects a 200 return code, but actually you're implementing a redirect. I tweaked the assertion accordingly.

3. I also assert that the thinker is updated as we expect. I have to `reload` our thinker instance to do this, since otherwise our in-memory `@thinker` won't have the latest changes to the database.

4. Because this test had a few more lines of code than others, I added some blank lines. I typically distinguish between 3 phases in a typical test: Arrange, Act and Assert. I've grouped this test into those 3 phases, which makes it clearer what we're actually testing (the `put` request).